tests/scenarios/template-tag-codemod-test.ts

```diff
@@ -0,0 +1,48 @@
+import { readFileSync } from 'fs-extra';
+import { appScenarios } from './scenarios';
+import QUnit from 'qunit';
+import { join } from 'path';
+
+const { module: Qmodule, test } = QUnit;
+
+appScenarios
+  .only('release')
+  .map('template-tag-codemod', project => {
+    project.mergeFiles({
+      app: {
+        components: {
+          'face.hbs': `<h1> this is a gjs file</h1>`,
+        },
+      },
+      'ember-cli-build.js': `'use strict';
+
+const EmberApp = require('ember-cli/lib/broccoli/ember-app');
+
+module.exports = function (defaults) {
+  const app = new EmberApp(defaults, {
+    // Add options here
+  });
+  return require('@embroider/compat').templateTagCodemod(app, {});
+};`,
+    });
+  })
+  .forEachScenario(async scenario => {
+    Qmodule(`${scenario.name}`, function (/* hooks */) {
+      test('running the codemod works', async function (assert) {
+        let app = await scenario.prepare();
+        await app.execute('node ./node_modules/ember-cli/bin/ember b');
+
+        // TODO figure out how to get assert.codeContains to understand template tag
+        const fileContents = readFileSync(join(app.dir, 'app/components/face.gjs'), 'utf-8');
+        assert.equal(
+          fileContents,
+          `export default <template>
+	<h1> this is a gjs file</h1>
+</template>;`
+        );
+        // TODO figure out how to get around the protection in place to not delete unversioned files
```

Comment:
this brings me back to what I said during the call. I imagine having a remove function that verifies that the file is in the repo (maybe after following links) before it tries to delete anything would be good enough for most people 🤔

Reply:
anything can happen between the time the function verifies that, and actual actioning of the remove opration, git rm handles all of that very nicely for us which is perfectly.

even if we had a function to test it's within the repo, what the current implementation does is not remove anything that isn't backed up (read source controlled) which is WAY safer than removing things that happen to be in the repo. and only creates things that don't already exist.

I always error on the side of safer when it comes to removing things on behalf of users and in this case it's the test that can't cope with the solution, so we shouldn't implement a lesser solution so the test is happier

Reply:
how about inverting it? make it have an optional config that you can pass a remove function to and we can add it to the docs that if you want it to do a git remove for **reasons** then you just pass a function that does git remove

thoughts?

Reply:
configuration is to be avoided if possible.
convention is that code mod does the right thing by default

if anything we need to make scenario tester source control aware/capable